xarray/core/variable.py

```diff
@@ -2001,15 +2001,15 @@ def coarsen(self, windows, func, boundary="exact", side="left", **kwargs):
         if not windows:
             return self.copy()
 
-        reshaped, axes = self._coarsen_reshape(windows, boundary, side)
+        reshaped, axes = self._coarsen_reshape(windows, boundary, side, **kwargs)
         if isinstance(func, str):
             name = func
             func = getattr(duck_array_ops, name, None)
             if func is None:
                 raise NameError(f"{name} is not a valid method.")
```

Comment:
Let's parse `keep_attrs` here and pass an appropriate `attrs` kwarg to `self._replace` below.